compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -533,3 +543,133 @@ private val PropertyDescriptor.declaresDefaultValue: Boolean
+        it.shortName.getShortName() == annotationKClass.simpleName && it.annotationType.resolve().declaration
+            .qualifiedName?.asString() == annotationKClass.qualifiedName
+    }.map { it.toAnnotation(annotationKClass.java) }
+}
+
+@KspExperimental
+fun <T : Annotation> KSAnnotated.isAnnotationPresent(annotationKClass: KClass<T>): Boolean =
+    getAnnotationsByType(annotationKClass).firstOrNull() != null
+
+@Suppress("UNCHECKED_CAST")
+private fun <T : Annotation> KSAnnotation.toAnnotation(annotationClass: Class<T>): T {
+    return Proxy.newProxyInstance(
+        annotationClass.classLoader,
+        arrayOf(annotationClass),
+        createInvocationHandler(annotationClass)
+    ) as T
+}
+
+@Suppress("TooGenericExceptionCaught")
+private fun KSAnnotation.createInvocationHandler(clazz: Class<*>): InvocationHandler {
+    val cache = ConcurrentHashMap<Pair<Class<*>, Any>, Any>(arguments.size)
+    return InvocationHandler { proxy, method, _ ->
+        if (method.name == "toString" && arguments.none { it.name?.asString() == "toString" }) {
+            clazz.canonicalName +
+                arguments.map { argument: KSValueArgument ->
+                    // handles default values for enums otherwise returns null
+                    val methodName = argument.name?.asString()
+                    val value = proxy.javaClass.methods.find { m -> m.name == methodName }?.invoke(proxy)
+                    "$methodName=$value"
+                }.toList()
+        } else {
+            val argument = try {
+                arguments.first { it.name?.asString() == method.name }
+            } catch (e: NullPointerException) {
+                throw IllegalArgumentException("This is a bug using the default KClass for an annotation", e)
+            }
+            when (val result = argument.value ?: method.defaultValue) {
+                is Proxy -> result
+                is ArrayList<*> -> {
+                    val value = { result.asArray(method) }
+                    cache.getOrPut(Pair(method.returnType, result), value)
+                }
+                else -> {
+                    when {
+                        method.returnType.isEnum -> {
+                            val value = { result.asEnum(method.returnType) }
+                            cache.getOrPut(Pair(method.returnType, result), value)
+                        }
+                        method.returnType.isAnnotation -> {
+                            val value = { (result as KSAnnotation).asAnnotation(method.returnType) }
+                            cache.getOrPut(Pair(method.returnType, result), value)
+                        }
+                        method.returnType.name == "java.lang.Class" -> {
+                            val value = { (result as KSType).asClass() }
+                            cache.getOrPut(Pair(method.returnType, result), value)
+                        }
+                        method.returnType.name == "byte" -> {
```

Comment:
the values that are returned for the Byte values are in the form an Integer and need to be narrowcast explicitly otherwise a ClassCastException occurs. The other values don't require casting so this isn't an issue and can be returned without casting.

Reply:
Thanks for the investigation. Does [short](https://kotlinlang.org/docs/basic-types.html) need to be handled as well?

Reply:
good catch, missed that type. 

To add more clarification, to the casting issues. It seems that this is an issue really only for the Annotations written in Java. Kotlin provides the correct class, however Java provides an Integer.

Reply:
Add an update and addressed the missing Short value